test/e2e/parallel/spacerequest_test.go

```diff
@@ -27,42 +34,48 @@ func TestCreateSpaceRequest(t *testing.T) {
 		// when
 		targetClusterRoles := []string{cluster.RoleLabel(cluster.Tenant)}
 		spaceRequest, parentSpace := CreateSpaceRequest(t, awaitilities, memberAwait.ClusterName,
-			WithSpecTierName("appstudio"),
+			WithSpecTierName("appstudio-env"),
 			WithSpecTargetClusterRoles(targetClusterRoles))
 
 		// then
 		// check for the subSpace creation
 		subSpace, err := awaitilities.Host().WaitForSubSpace(t, spaceRequest.Name, spaceRequest.Namespace, parentSpace.GetName(),
 			UntilSpaceHasTargetClusterRoles(targetClusterRoles),
-			UntilSpaceHasTier("appstudio"),
+			UntilSpaceHasTier("appstudio-env"),
 			UntilSpaceHasAnyProvisionedNamespaces(),
 		)
 		require.NoError(t, err)
 		subSpace, _ = VerifyResourcesProvisionedForSpace(t, awaitilities, subSpace.Name, UntilSpaceHasAnyTargetClusterSet())
 		spaceRequest, err = memberAwait.WaitForSpaceRequest(t, types.NamespacedName{Namespace: spaceRequest.GetNamespace(), Name: spaceRequest.GetName()},
 			UntilSpaceRequestHasConditions(Provisioned()),
-			UntilSpaceRequestHasStatusTargetClusterURL(memberCluster.Spec.APIEndpoint))
+			UntilSpaceRequestHasStatusTargetClusterURL(memberCluster.Spec.APIEndpoint),
+			UntilSpaceRequestHasNamespaceAccess(),
+		)
 		require.NoError(t, err)
+		// verify secret containing kubeconfig was provisioned
+		kubeClient, namespaceAccessSecret := newKubeClientFromSecret(t, memberAwait, spaceRequest.Namespace, spaceRequest.Status.NamespaceAccess[0].SecretRef)
+		// check that kubeconfig is valid
+		validateKubeClient(t, kubeClient, spaceRequest)
```

Comment:
couldn't we just run this as part of the `UntilSpaceRequestHasNamespaceAccess()` function?

Reply:
it was not nice indeed, so I ended up doing the following in https://github.com/codeready-toolchain/toolchain-e2e/pull/706/commits/3f315e55a25920ece38c2351288e4faa2176ae66:
- `UntilSpaceREquestHasNamespaceAccess` validates the number of namespaces and names in `spacerequest.Status.NamespaceAccess` are matching the ones from the `subspace.Status.ProvisionedNamespaces` 
- `VerifyNamespaceAccessForSpaceRequest` is a new function which validates the all the entries `spacerequest.Status.NamespaceAccess` are valid kubeconfig files

I had to split the validation since the `Until...` functions are meant for matching values or comparing number of elements and it's a bit more tricky where you have to do error handling, especially reporting those in the diff. Hope this looks better now.